src/irmin-pack/pack.ml

```diff
@@ -314,9 +315,11 @@ struct
             (IO.name t.pack.block)
         in
         t.pack.block <- block;
-        Dict.sync t.pack.dict)
-      else Dict.sync t.pack.dict;
-      Index.sync t.pack.index
+        Dict.sync t.pack.dict;
+        Index.sync t.pack.index)
```

Comment:
Why wasn't `Index.sync` called here before?

Reply:
I think it was on L319, and this is a case of OCamlformat / OCaml syntax not making that clear.